datadog_checks_base/datadog_checks/base/checks/openmetrics/mixins.py

```diff
@@ -461,28 +485,31 @@ def process_metric(self, metric, scraper_config, metric_transformers=None):
         try:
             self.submit_openmetric(scraper_config['metrics_mapper'][metric.name], metric, scraper_config)
         except KeyError:
-            if metric_transformers is not None:
-                if metric.name in metric_transformers:
-                    try:
-                        # Get the transformer function for this specific metric
-                        transformer = metric_transformers[metric.name]
-                        transformer(metric, scraper_config)
-                    except Exception as err:
-                        self.log.warning("Error handling metric: {} - error: {}".format(metric.name, err))
-                else:
-                    self.log.debug(
-                        "Unable to handle metric: {0} - error: "
-                        "No handler function named '{0}' defined".format(metric.name)
-                    )
-            else:
-                # build the wildcard list if first pass
-                if scraper_config['_metrics_wildcards'] is None:
-                    scraper_config['_metrics_wildcards'] = [x for x in scraper_config['metrics_mapper'] if '*' in x]
-
-                # try matching wildcard (generic check)
-                for wildcard in scraper_config['_metrics_wildcards']:
-                    if fnmatchcase(metric.name, wildcard):
-                        self.submit_openmetric(metric.name, metric, scraper_config)
+            if metric_transformers is not None and metric.name in metric_transformers:
+                try:
+                    # Get the transformer function for this specific metric
+                    transformer = metric_transformers[metric.name]
+                    transformer(metric, scraper_config)
+                except Exception as err:
+                    self.log.warning('Error handling metric: {} - error: {}'.format(metric.name, err))
+
+                return
+
+            # build the wildcard list if first pass
+            if scraper_config['_metrics_wildcards'] is None:
+                scraper_config['_metrics_wildcards'] = [x for x in scraper_config['metrics_mapper'] if '*' in x]
+
+            # try matching wildcard (generic check)
```

Comment:
```suggestion
            # try matching wildcard
```
we might want to document the change of behavior as well, if people had wildcards in custom checks alongs metrics mapper they could send additional custom metrics without paying attention before we weren't considering those before